core/rawdb/celo_accessors_test.go

```diff
@@ -0,0 +1,131 @@
+package rawdb
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestPreGingerbreadAdditionalFields tests ReadPreGingerbreadAdditionalFields function
+func TestPreGingerbreadAdditionalFields(t *testing.T) {
+	db := NewMemoryDatabase()
+
+	mockData, err := rlp.EncodeToBytes(&PreGingerbreadFields{
+		BaseFee:  big.NewInt(1000),
+		GasLimit: big.NewInt(2000),
+	})
+	assert.NoError(t, err)
+
+	type SeedData struct {
+		hash common.Hash
+		data []byte
+	}
+
+	tests := []struct {
+		name         string
+		seedData     []SeedData
+		hash         common.Hash
+		expectedRes  *PreGingerbreadFields
+		returnsError bool
+	}{
+		{
+			name:         "should return nil if data is not found",
+			seedData:     []SeedData{},
+			hash:         common.HexToHash("0x1"),
+			expectedRes:  nil,
+			returnsError: false,
+		},
+		{
+			name: "should return data",
+			seedData: []SeedData{
+				{
+					hash: common.HexToHash("0x2"),
+					data: mockData,
+				},
+			},
+			hash: common.HexToHash("0x2"),
+			expectedRes: &PreGingerbreadFields{
+				BaseFee:  big.NewInt(1000),
+				GasLimit: big.NewInt(2000),
+			},
+			returnsError: false,
+		},
+		{
+			name: "should return error if data is broken",
+			seedData: []SeedData{
+				{
+					hash: common.HexToHash("0x3"),
+					data: []byte{0x1, 0x2, 0x3},
+				},
+			},
+			hash:         common.HexToHash("0x3"),
+			expectedRes:  nil,
+			returnsError: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			for _, seed := range test.seedData {
+				err := db.Put(preGingerbreadAdditionalFieldsKey(seed.hash), seed.data)
+				assert.NoError(t, err)
```

Comment:
I would use require here instead of assert. A failed assert statement does not abort the test, whereas require will. In this case if the insertion fails then there is no reason to continue the test.

Reply:
Learned new best practice! Thank you :) I will fix that

Reply:
Most of `assert.*` have been replaced with `require.*`. Resolved the following related comments.